File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int n = strs.size();
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> hash;
        for(int i=0; i<n; i++){
            string curr = strs[i];
            sort(curr.begin(), curr.end());
            hash[curr].push_back(strs[i]);
            // if(hash.find(strs[i]) != hash.end()) hash[curr].push_back(strs[i]);
            // else{
            //     hash[curr].push_back(strs[i]);
            // }
        }

        for(auto it: hash){
            ans.push_back(it.second);
        }

        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const n = strs.length;
    let hash = new Map();

    for(let i=0; i<n; i++){
        const sortedStr = strs[i].split('').sort().join('');

        if(!hash.has(sortedStr)){
            hash.set(sortedStr, []);
        }

        hash.get(sortedStr).push(strs[i]);
    }

    return Array.from(hash.values());
};